### PR TITLE
[Tyr] Upgrade endpoints '/' & add '/status'

### PR DESCRIPTION
**Pull request by mbounabi**
https://jira.kisio.org/browse/NAVP-1305
Change response of the following endpoints:

- /:
will return only root-level endpoints. 
```                                               
{
    "api": {
        "href": "http://localhost:5000/v0/api/"
    },
    "autocompleteparameter": {
        "href": "http://localhost:5000/v0/autocomplete_parameters/"
    },
    "billingplan": {
        "href": "http://localhost:5000/v0/billing_plans/"
    },
    "bssprovider": {
        "href": "http://localhost:5000/v0/bss_providers"
    },
    "cities": {
        "href": "http://localhost:5000/v0/cities/"
    },
    "citiesstatus": {
        "href": "http://localhost:5000/v0/cities/status"
    },
    "end_points": {
        "href": "http://localhost:5000/v0/end_points/"
    },
    "equipmentsprovider": {
        "href": "http://localhost:5000/v0/equipments_providers"
    },
    "index": {
        "href": "http://localhost:5000/"
    },
    "instance": {
        "href": "http://localhost:5000/v0/instances/"
    },
    "jobs": {
        "href": "http://localhost:5000/v0/jobs/"
    },
    "user": {
        "href": "http://localhost:5000/v0/users/"
    }
}
```

- /status:
```
{
    "db version": "35d86bbbe642"
}
```

### CODE REVIEW

**Review by benoit-bst — APPROVED** [review]
dB version is honey :bee:

File: source/tyr/tyr/resources.py [hunk 1]
```diff
@@ -63,26 +63,41 @@
...
+            try:
+                tyr_db.connect()
+                result = tyr_db.execute("SELECT version_num FROM alembic_version")
+                for row in result:
+                    return row['version_num']
+            except Exception as e:
+                logging.exception("Tyr db not reachable : {}".format(e.message))
+                return None
+
+        return {'db version': check_db(), 'api': marshal(models.Api.query.all(), api_fields)}
```
Comment (kinnou02) [inline source/tyr/tyr/resources.py]: this isn't related, I would have put the db version in a `/status`

Comment (mbounabi) [reply source/tyr/tyr/resources.py]: When I suggested to put it in /api in Slack, you reacted with thumbs up, as there's currently no endpoint /status.
If it is to be created, I suggest to do it in an other PR

**Review by benoit-bst — APPROVED** [review]
OKay, so now **/api** a thin output :speak_no_evil:

**Comment by mbounabi** [comment]
> OKay, so now **/api** a thin output speak_no_evil

The forgotten endpoint...

File: source/tyr/tyr/resources.py [hunk 2]
```diff
@@ -82,7 +80,29 @@ def get(self):
...
+                response[endpoint] = {'href': url_for(endpoint, _external=True)}
+            except werkzeug.routing.BuildError:
+                logging.warning('Could not build url for endpoint \'{}\' '.format(endpoint))
+        return response
+
+
+class Status(flask_restful.Resource):
+    def get(self):
+        def check_db():
+            tyr_db = sqlalchemy.create_engine(current_app.config['SQLALCHEMY_DATABASE_URI'])
```
Comment (kinnou02) [inline source/tyr/tyr/resources.py]: This look dangerous, you are creating a new connection pool to the database, that may or may not be closed.
You should take some inspiration from kirin's status: https://github.com/CanalTP/kirin/blob/master/kirin/resources.py#L47-L55

**Review by kinnou02 — CHANGES_REQUESTED** [review]

**Comment by kadhikari** [comment]
A little test on new end point is always welcome

File: source/tyr/tyr/resources.py (hunk 2 above)
Comment (mbounabi) [reply source/tyr/tyr/resources.py]: Not really the same case, as Kirin queries a table defined in the model. The table 'alembic_version' is automatically generated.
But it does need to be changed in some way

**Review by kinnou02 — APPROVED** [review]
I agree with @kadhikari, apart from that  it's good

**Review by datanel — APPROVED** [review]

**Review by benoit-bst — APPROVED** [review]

**Review by kadhikari — APPROVED** [review]

**Pull request merged**